x-pack/plugins/osquery/public/saved_queries/use_update_saved_query.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation, useQueryClient } from 'react-query';
+import { i18n } from '@kbn/i18n';
+
+import { useKibana } from '../common/lib/kibana';
+import { savedQuerySavedObjectType } from '../../common/types';
+import { PLUGIN_ID } from '../../common';
+import { pagePathGetters } from '../common/page_paths';
+import { SAVED_QUERIES_ID } from './constants';
+import { useErrorToast } from '../common/hooks/use_error_toast';
+
+interface UseUpdateSavedQueryProps {
+  savedQueryId: string;
+}
+
+export const useUpdateSavedQuery = ({ savedQueryId }: UseUpdateSavedQueryProps) => {
+  const queryClient = useQueryClient();
+  const {
+    application: { navigateToApp },
+    savedObjects,
+    security,
+    notifications: { toasts },
+  } = useKibana().services;
+  const setErrorToast = useErrorToast();
+
+  return useMutation(
+    async (payload) => {
+      const currentUser = await security.authc.getCurrentUser();
+
+      if (!currentUser) {
+        throw new Error('CurrentUser is missing');
+      }
+
+      return savedObjects.client.update(savedQuerySavedObjectType, savedQueryId, {
+        // @ts-expect-error update types
+        ...payload,
+        updated_by: currentUser.username,
+        updated_at: new Date(Date.now()).toISOString(),
```

Comment:
just a nit, but wouldn't this be equivalent to `new Date().toISOString()` since the date object defaults to the current time?